### DISCUSSION TITLE
can anyone tell me why i fail the automated tests pull request #8933

### DISCUSSION THREAD

**Discussion by samarth64** [TheAlgorithms/Python#8934] [Q&A] [answered]
I have recently started contributing to GitHub so I don't how the automated test works so I don't know what error or problem is that it fails to merge 
help me

**Comment by tianyizheng02** [comment, marked as answer]
<img width="850" alt="screenshot" src="https://github.com/TheAlgorithms/Python/assets/52298854/acde2f50-07e7-41c1-9491-714ab34a56a2">

You can click on "Details" to see why an automated test is failing.

In short, ruff is failing because we recently updated it to a newer version, and that caused ruff to suddenly consider many things wrong that wasn't considered wrong before. You're not at fault here, and these ruff errors will be fixed in a future PR.

The build is failing because one of your changes caused a unit test to fail, meaning that your change caused a function to produce the wrong output. You'll need to fix this.
